### PR TITLE
Add dashboard link to view to emit user list as Mailchimp-compatible CSV

### PR DESCRIPTION
**Pull request by waxlamp**
This is needed to automate a somewhat tricky process when done by hand. The resulting CSV can be used to import the users to a Mailchimp account in order to keep the Mailchimp audience accurate when sending (infrequent) mass emails.

### CODE REVIEW

**Comment by kabilar** [comment]
Thank you @waxlamp.

**Comment by mvandenburgh** [comment]
We already have this feature as a Django admin action (code is [here](https://github.com/dandi/dandi-archive/blob/master/dandiapi/api/admin.py#L66-L72)), which IMO is more appropriate than a management command for this particular action. I'm not aware of what the driving use case is behind this, though. Any thoughts @waxlamp (or anyone else)?

**Comment by waxlamp** [comment]
> We already have this feature as a Django admin action (code is [here](https://github.com/dandi/dandi-archive/blob/master/dandiapi/api/admin.py#L66-L72)), which IMO is more appropriate than a management command for this particular action. I'm not aware of what the driving use case is behind this, though. Any thoughts @waxlamp (or anyone else)?

Thanks Mike, I hadn't realized that (obviously).

The existing functionality doesn't cover the new use case, which is to export a CSV file consisting of just the active users' email addresses, first, and last names. I'm sure we could extend the existing code to cover this as well. I'll reach out to you to discuss further.

File: dandiapi/api/management/commands/mailchimp_csv.py
Comment (asmacdo) [inline dandiapi/api/management/commands/mailchimp_csv.py:24]:
```suggestion
   #  Do not include the empty user ('', '', '')
    writer.writerows(d for d in data if d[0])
```

**Comment by waxlamp** [comment]
@mvandenburgh, attempt number two -- in the dashboard view, I added a "navbar" link to a new view that generates the necessary CSV. Please take a look and let me know if there's a better way to accomplish this. In particular, perhaps I don't need to manufacture a URL to the new view and can somehow use the view function directly, rather than by reversing the named view? Or really any other improvements.

File: dandiapi/api/views/dashboard.py
Comment (danlamanna) [inline dandiapi/api/views/dashboard.py:93]: I would use [StreamingHttpResponse](https://docs.djangoproject.com/en/5.0/ref/request-response/#streaminghttpresponse-objects) if possible.

File: dandiapi/api/views/dashboard.py
Comment (danlamanna) [inline dandiapi/api/views/dashboard.py:91]: Maybe use [.iterator](https://docs.djangoproject.com/en/5.0/ref/models/querysets/#iterator).

File: dandiapi/api/views/dashboard.py
Comment (danlamanna) [inline dandiapi/api/views/dashboard.py:99]: Now I'm being pedantic, but I find [csv.DictWriter](https://docs.python.org/3/library/csv.html#csv.DictWriter) to be harder to misuse.

File: dandiapi/api/views/dashboard.py
Comment (waxlamp) [reply dandiapi/api/views/dashboard.py]: Done in 88d9798884bedd0652f4b9a9ce605b8b4a1de893.

File: dandiapi/api/views/dashboard.py
Comment (waxlamp) [reply dandiapi/api/views/dashboard.py]: Done in e2808d15baf4643352924eaecffea9643a26d9af.

File: dandiapi/api/views/dashboard.py
Comment (mvandenburgh) [inline dandiapi/api/views/dashboard.py:90]:
```suggestion
    users = User.objects.filter(metadata__status=UserMetadata.Status.APPROVED).exclude(email='')
```

The built-in `is_active` field is used to determine if the user can login or not. A PENDING user (i.e. a user waiting for approval) is still allowed to login, they just can't do most operations, so they are actually have `is_active=True`.

File: dandiapi/api/views/dashboard.py
Comment (mvandenburgh) [inline dandiapi/api/views/dashboard.py:93]:
```suggestion
    def streaming_output() -> Iterator[tuple[str, str, str]]:
```

File: dandiapi/urls.py
Comment (mvandenburgh) [inline dandiapi/urls.py:109]:
```suggestion
    path('dashboard/mailchimp/', mailchimp_csv_view, name='mailchimp-csv'),
```

Django will automatically redirect, but just to be consistent with the rest of the routes.

**Review by mvandenburgh — CHANGES_REQUESTED** [review]

File: dandiapi/api/views/dashboard.py
Comment (waxlamp) [reply dandiapi/api/views/dashboard.py:95]: IIUC, this generator yields strings. The reason we're using that `Echo` pseudo-buffer thing is to cause the  `writeheader()` and `writerow()` methods to return the string that they write. (Apparently, those methods just return the result of calling `.write()` on the underlying filelike object. By contrast, if we did this with a real filelike, whose `.write()` method returns the number of bytes written, we'd end up sending those byte counts to the streaming response.) It's sort of a mess 🫤 but it works.

If my analysis is correct, I'll update the suggestion to use `Iterator[str]` and commit it. Let me know what you think.

Comment (mvandenburgh) [reply dandiapi/api/views/dashboard.py:95]: Ah you're right, I verified it locally and it is indeed a `Iterator[str]`.

**Review by mvandenburgh — APPROVED** [review]

**Pull request merged**
